container-registry-configuration/README.md

```diff
@@ -0,0 +1,189 @@
+# Container Registry Configuration
+
+The purpose of this RFC is to specify how we configure Kubernetes clusters for container registries.
+
+## Problem Statement
+
+Docker Hub has [download rate limit](https://docs.docker.com/docker-hub/download-rate-limit/) and it is pretty strict. To use Docker Hub without authentication is not a livable option for us. We started to hit the limit in big CAPI clusters during upgrades.
+
+## Current status
+
+- We use images from different registries. Here is the summary of registry usage of a test MC.
```

Comment:
I want to mention _pull through caching_ since it's not listed in this PR.

AWS ECR, as the only larger cloud registry, [supports](https://docs.aws.amazon.com/AmazonECR/latest/userguide/pull-through-cache.html) it already. When Docker rate limits became enforced, I experienced the same problems in large business Kubernetes installations. It was worked around by

- [AWS public ECR images](https://gallery.ecr.aws/) wherever available (major ones like Ubuntu, Flux, etc. seem available) because AWS yet has no rate limit or at least a per-account limit that doesn't totally kill productivity (example: `MY_ACCOUNT_ID.dkr.ecr.eu-central-1.amazonaws.com/ecr-public/docker/library/ubuntu:20.04` instead of `ubuntu:20.04`)
- rewriting images after enabling the pass-through cache (replace `docker.io/...` with AWS ECR image, for instance, which is quite some effort if not automated)

It's a _valid workaround_ to buy time and avoid live issues. I fully agree to buying into container registries such as Harbor which can work around the limitations from a centrally-configured place.

Anecdotal, minor story: replacing `image: abc` with `image: def` seems not uncommon. In addition to the above case, when I tried IPv6-only clusters for my private purposes, I noticed that lots of registries (e.g. `ghcr.io`) and websites (e.g. GitHub release binaries) aren't IPv6-reachable and I therefore patched a few of my Helm chart installations, for example like `redis: { image: { repository: registry.ipv6.docker.com/library/redis } }`. A container registry should have support for such patches.